solidity/contracts/AltheaDB.sol

```diff
@@ -1,54 +1,417 @@
 //SPDX-License-Identifier: Apache-2.0
 pragma solidity 0.8.21; // Force solidity compliance
 
+/// Identity struct
+struct Identity {
+    uint128 mesh_ip;
+    uint256 wg_key;
+    address eth_addr;
+}
+
+/// Identity struct for an exit, including a list of allowed region codes
+/// This is to protect a user from connecting to an exit that does not allow
+/// their region despite providing the best connection metrics otherwise
+/// The payment types specification prevents clients from moving to exits
+/// that do not accept their tokens as payment
+struct ExitIdentity {
+    uint128 mesh_ip;
+    uint256 wg_key;
+    address eth_addr;
+    uint256[] allowed_regions;
```

Comment:
Unless you're going to use strings for region values, uint256 seems a bit excessive

Reply:
it is excessive, we could probably do u32 but I don't see how extra options hurts us here.